### DISCUSSION TITLE
Moose Representation for Normal Derivatives and Directional Derivatives

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#25015] [Q&A General] [answered]
Some codes allow for access to the values for the normal derivative in the following way, nx, ny, etc., and directional derivatives as ux, px, uy, etc.. How can these values be applied for use in MOOSE, and what is their syntax?

**Comment by GiudGiud** [comment, marked as answer]
Hello

so normal near a boundary?

for directional derivative you can retrieve a gradient using coupledGradient(“name of variable parameter”)

for normal derivatives, use the gradient and the _normals attribute defined near boundaries only 

Guillaume

**Reply by TLWise** [reply]
Thank you for the guidance.
